### PR TITLE
Pay with PayPal: Add alternative shortcode output for email notifications

### PR DESCRIPTION
**Pull request by mmtr**
Part of https://github.com/Automattic/jetpack/issues/17161
Fixes 190-gh-dotcom-manage

#### Changes proposed in this Pull Request:
The `simple-payment` shortcode renders a PayPal button that is initialized with a JS file. However, that doesn't work when rendered in a email, so this PR changes the output of the shortcode when used on an email notification. In those cases, we simply display now a textual call-to-action that links to the post so folks can purchase from the website directly.

Before | After
--- | ---
Screen Shot 2020-09-16 at 15 38 16 | Screen Shot 2020-09-16 at 15 39 49

#### Jetpack product discussion
pbAPfg-OF-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

WordPress.com
- Switch to a site under a Premium plan.
- Publish a post that contains a "Pay with PayPal" block.
- Grab the IDs of the post and the site where the post has been published.
- In a WP.com sandbox run `php ./bin/subscriptions/send.php <BLOG_ID> post <POST_ID> <YOUR_EMAIL>`. 
- Check the email received and make sure it contains a "Visit the site to purchase" link replacing the non-functional PayPal button.
- Make sure the link redirects to the published post.
- Make sure the PayPal button is visible on the published post.

Jetpack
- [Spin a JN site running this branch](http://jurassic.ninja/create/?jetpack-beta&branch=fix/pay-with-paypal-fallback-context).
- Set up Jetpack and continue using the Free plan.
- Add a Jetpack Premium to the JN site plan through Store Admin.
- Subscribe to the site using the [Subscriptions module](https://jetpack.com/support/subscriptions/):
  - Go to Jetpack > Settings > Discussion.
  - Activate the "Let visitors subscribe to new posts and comments via email" option.
  - Visit an already published post (the "Hello World!" post automatically created by JN is fine).
  - Subscribe to the site by leaving a comment and checking the "Notify me of new posts by email." option.
  - Confirm the subscription by clicking on the confirmation email you've received.
- Publish a post that contains a "Pay with PayPal" block.
- Check the email subscription and make sure it contains a "Visit the site to purchase" link replacing the non-functional PayPal button.
- Make sure the link redirects to the published post.
- Make sure the PayPal button is visible on the published post.

#### Proposed changelog entry for your changes:
Pay with PayPal: Add alternative shortcode output for email notifications

### CODE REVIEW

File: modules/simple-payments/simple-payments.php
Comment (jeherve) [inline modules/simple-payments/simple-payments.php:244]:
```suggestion
				'<a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s</a>',
```

**Review by jeherve — CHANGES_REQUESTED** [review]

**Review by jeherve — COMMENTED** [review]
Would it be possible to find a solution that works for both WordPress.com simple sites and other Jetpack sites out there? In the past, we've implemented things like r213058-wpcom or r171528-wpcom to customize the content of email notifications. Could we do something similar here?

**Comment by mmtr** [comment]
> Would it be possible to find a solution that works for both WordPress.com simple sites and other Jetpack sites out there?

Maybe the current solution works for both? I think emails subscriptions for Jetpack sites are also sent from async jobs running in WP.com and using the `Subscription_Mailer` class (which is what the `\A8C\Display_context` library needs to determine if a post is being rendered in an email).

Let me double check that and confirm if that's really the case.

**Comment by mmtr** [comment]
@jeherve I'm trying to confirm the above, but I'm having a hard time finding a way of making the Pay with PayPal block as available. The block requires a Premium plan but it seems that plan is no longer available (there are new plans now: Security and Complete), so the block is never registered. 

I guess the availability of the premium blocks has not been updated yet to take into account the new plans. Is there any way I can force the old plans to appear when upgrading?

**Comment by jeherve** [comment]
Yes, the new plans make this complicated. We're fixing this in #17177, but cannot merge it just yet because the e2e tests also need fixing to account for the new plans.

To test this I'd recommend using an existing site with a Premium Plan, or adding a Premium plan via Store Admin via your test site.

**Comment by mmtr** [comment]
I could confirm Jetpack emails subscriptions are sent from WP.com via the `Subscriber_Mailer` class, so this fix should cover both WP.com simple sites and Jetpack sites.

However I couldn't test the fix on a Jetpack site because the email is always sent through an async job from production, where the companion patch D49660-code is not available yet. I tried sandboxing everything (API, setting my sandbox address in `JETPACK__SANDBOX_DOMAIN`) but couldn't find a way of sending the email from my sandbox.

I guess we could merge D49660-code before merging this PR if we really want to verify the fix on a Jetpack site, but that will cause the WP.com/Jetpack codebases to be temporary out of sync.

**Comment by jeherve** [comment]
> I guess we could merge D49660-code before merging this PR if we really want to verify the fix on a Jetpack site, but that will cause the WP.com/Jetpack codebases to be temporary out of sync.

Do you think we could give it a try? We're blocked from merging here in Jetpack for now anyway, because of the e2e test issues I mentioned earlier.

**Comment by kwight** [comment]
I've deployed D49660-code, but haven't been able to get an email to show up automatically via production with a newish JN site.

First, I had problems subscribing: I think heavy caching means the JN site doesn't show up under my `https://wordpress.com/following/manage`. I thought I managed to get all the right settings done by API with the developer console (`WP.COM API POST v1.1/read/following/mine/new` and `WP.COM API POST v1.2/read/site/implicit-aardwolf.jurassic.ninja/post_email_subscriptions/new`), but posting a new post hasn't resulted in a subscriber email, so I also haven't truly verified the behaviour we want. FWIW, I was able to use the `send.php` script with that JN site and see what we want, but... not the same. 🤔

**Comment by mmtr** [comment]
Just confirmed it works 🎉 . This is subscriber email I received after publishing a post in my Jurassic Ninja site:

Screen Shot 2020-09-18 at 12 22 37

Testing instructions for Jetpack sites:
- [Spin a JN site running this branch](http://jurassic.ninja/create/?jetpack-beta&branch=fix/pay-with-paypal-fallback-context).
- Set up Jetpack and continue using the Free plan.
- Add a Jetpack Premium to the JN site plan through Store Admin.
- Subscribe to the site using the [Subscriptions module](https://jetpack.com/support/subscriptions/):
  - Go to Jetpack > Settings > Discussion.
  - Activate the "Let visitors subscribe to new posts and comments via email" option.
  - Visit an already published post (the "Hello World!" post automatically created by JN is fine).
  - Subscribe to the site by leaving a comment and checking the "Notify me of new posts by email." option.
  - Confirm the subscription by clicking on the confirmation email you've received.
- Publish a post that contains a Pay with PayPal block.
- Check the email subscription and make sure it contains a "Visit the site to purchase" link replacing the non-functional PayPal button.

**Review by jeherve — APPROVED** [review]

**Pull request merged**
